src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -134,11 +167,11 @@ private static void EmitRegexMethod(IndentedTextWriter writer, RegexMethod rm, s
             writer.Write("    public static global::System.Text.RegularExpressions.Regex Instance { get; } = ");
 
             // If we can't support custom generation for this regex, spit out a Regex constructor call.
-            if (!SupportsCustomCodeGeneration(rm))
+            if (!SupportsCodeGeneration(rm))
             {
                 writer.WriteLine($"new global::System.Text.RegularExpressions.Regex({patternExpression}, {optionsExpression}, {timeoutExpression});");
                 writer.WriteLine("}");
-                return;
+                return ImmutableArray.Create(Diagnostic.Create(DiagnosticDescriptors.LimitedSourceGeneration, rm.MethodSyntax.GetLocation()));
```

Comment:
do we have the offset in the pattern where we had to bail to include in this message? I assume the location here is just the line in the generated source.

Reply:
> I assume the location here is just the line in the generated source.

The location here is the user-written [RegexGenerator] partial method.

> do we have the offset in the pattern where we had to bail to include in this message?

We don't currently have a way to track where in the pattern a RegexNode came from.  Most of these also won't be specific to the an offset into the pattern, e.g. it'll be because of the RegexOptions specified, or because the whole expression was too complex, etc.

Reply:
What about the case where the expression reached maxtree depth? That is something we'll know, the question only is if knowing that could help the user change the code in order to be able to benefit from source generation.